torch/csrc/distributed/Module.cpp

```diff
@@ -338,6 +367,271 @@ PyObject* THDPModule_recv(PyObject *_unused, PyObject *args)
   END_HANDLE_TH_ERRORS
 }
 
+
+PyObject* THDPModule_allReduceMultiGPU(PyObject *_unused, PyObject *args)
+{
+  HANDLE_TH_ERRORS
+  std::vector<at::Tensor> descriptors;
+  Py_ssize_t tmp_length;
+  std::size_t length;
+  THDGroup group;
+  THDReduceOp op;
+  PyObject* sequence;
+
+  if (PyTuple_GET_SIZE(args) != 3) {
+    goto invalid_arguments;
+  }
+
+  if (!PySequence_Check(PyTuple_GET_ITEM(args, 0))) {
+    goto invalid_arguments;
+  }
+
+  sequence = PySequence_Fast(PyTuple_GET_ITEM(args, 0), nullptr);
+  if (!sequence) {
+    goto invalid_arguments;
+  }
+
+  tmp_length = PySequence_Fast_GET_SIZE(sequence);
+  THPUtils_assert(tmp_length >= 0, "couldn't obtain the length of %s",
```

Comment:
nit: this cannot fail now, so no need for error checking. You can safely assign to `length` directly.